docs/design/configuration.md

```diff
@@ -0,0 +1,141 @@
+# Kubernetes Component Configuration
+
+## Abstract
+
+A generic ``Configuration`` object to store command-line flags that is used for
```

Comment:
Please do not describe it this way. That we used command-line flags to configure our components was repeating bad Google habits. We shouldn't be using command-line flags. I just made a comment to this effect on #1627.